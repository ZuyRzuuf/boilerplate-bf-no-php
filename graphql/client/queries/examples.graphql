query Examples {
    examples {
        ...examplesData
    }
}

fragment examplesData on Example {
    uuid
    name
}