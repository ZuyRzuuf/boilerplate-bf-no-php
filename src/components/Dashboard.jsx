import React from 'react';
import {Query} from 'react-apollo';

import Examples from '../../graphql/client/queries/examples.graphql';

const Dashboard = () => {
    return (
        <>
            <div>Dashboard</div>
            <Query query={Examples}>
                {({ loading, error, data }) => {
                    if (loading) return <p>Loading...</p>;

                    if (error) return <p>Error :(</p>;

                    return data.examples.map(({ uuid, name }) => (
                        <div key={uuid}>
                            <p>
                                {`${uuid}: ${name}`}
                            </p>
                        </div>
                    ));
                }}
            </Query>
        </>
    );
};

export default Dashboard;
