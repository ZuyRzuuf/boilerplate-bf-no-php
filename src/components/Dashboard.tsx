import * as React from 'react';
// import {Query} from 'react-apollo';

// import Examples from '../../graphql/client/queries/examples.graphql';

const Dashboard = () => {
    return (
        <>
            <div>Dashboard</div>
        </>
    );
};

export default Dashboard;
