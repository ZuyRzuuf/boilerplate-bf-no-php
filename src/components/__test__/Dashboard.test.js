import * as React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';

import Dashboard from '../Dashboard';

import * as Examples from '../../../graphql/client/queries/examples.graphql';

describe('<Dashboard /> suite', function() {
    const mocks = [
        {
            request: {
                query: Examples,
                variables: {},
            },
            result: {
                data: {
                    dog: { id: 'ee5ae76d-9b91-4270-a007-fad2054e2e75', name: 'example' },
                },
            },
        },
    ];

    it('renders correctly isolated component', () => {
        const wrapper = shallow(
            <MockedProvider addTypename={false} mocks={mocks}>
                <Dashboard />
            </MockedProvider>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly HTML tree', () => {
        const wrapper = shallow(
            <MockedProvider addTypename={false} mocks={mocks}>
                <Dashboard />
            </MockedProvider>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly components tree', () => {
        const wrapper = shallow(
            <MockedProvider addTypename={false} mocks={mocks}>
                <Dashboard />
            </MockedProvider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
