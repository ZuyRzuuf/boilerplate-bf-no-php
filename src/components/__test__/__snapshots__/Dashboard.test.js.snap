// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite renders correctly 1`] = `
<Fragment>
  <div>
    Dashboard
  </div>
  <Query
    query={
      Object {
        "Examples": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "Examples",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "examples",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "directives": Array [],
                          "kind": "FragmentSpread",
                          "name": Object {
                            "kind": "Name",
                            "value": "examplesData",
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
            Object {
              "directives": Array [],
              "kind": "FragmentDefinition",
              "name": Object {
                "kind": "Name",
                "value": "examplesData",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "uuid",
                    },
                  },
                  Object {
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "name",
                    },
                  },
                ],
              },
              "typeCondition": Object {
                "kind": "NamedType",
                "name": Object {
                  "kind": "Name",
                  "value": "Example",
                },
              },
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 119,
            "source": Object {
              "body": "query Examples {
    examples {
        ...examplesData
    }
}

fragment examplesData on Example {
    uuid
    name
}",
              "locationOffset": Object {
                "column": 1,
                "line": 1,
              },
              "name": "GraphQL request",
            },
            "start": 0,
          },
        },
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "Examples",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "examples",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "directives": Array [],
                        "kind": "FragmentSpread",
                        "name": Object {
                          "kind": "Name",
                          "value": "examplesData",
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
          Object {
            "directives": Array [],
            "kind": "FragmentDefinition",
            "name": Object {
              "kind": "Name",
              "value": "examplesData",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "uuid",
                  },
                },
                Object {
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                },
              ],
            },
            "typeCondition": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Example",
              },
            },
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 119,
          "source": Object {
            "body": "query Examples {
    examples {
        ...examplesData
    }
}

fragment examplesData on Example {
    uuid
    name
}",
            "locationOffset": Object {
              "column": 1,
              "line": 1,
            },
            "name": "GraphQL request",
          },
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Query>
</Fragment>
`;
